### PR TITLE
pass prompt state via URL parameter

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

**Comment by ericcf** [comment]
@eschlange clearly still lots of refactoring that could/should be done on this project. My only excuse is that we're a day late and a dollar short.

File: app/scripts/controllers/slide.js [hunk 1]
```diff
@@ -3,8 +3,14 @@
 
   angular.module('focusAppApp')
     .controller('SlideCtrl', function($scope, $sce, $routeParams, $window,
-                                      $timeout, lessonService, dlService,
-                                      formService, prService, guidService) {
+                                      $timeout, $location, lessonService,
+                                      dlService, formService, prService,
```
Comment (michaelwehrley) [inline app/scripts/controllers/slide.js]: I think you can remove `formService`.

File: app/scripts/controllers/main.js [hunk 2]
```diff
@@ -34,16 +38,14 @@
               prService.persistEncryptedString('prompt', '').execute();
 
               if (currentHour >= 9 && currentHour <= 12) {
-                $window.location = '#/slide?child_of=' + userData.window1 + '&construct=assessment1';
+                $window.location = promptLocationForWindow(userData.window1);
               } else if (currentHour >= 13 && currentHour <= 17) {
-                $window.location = '#/slide?child_of=' + userData.window2 + '&construct=assessment1';
+                $window.location = promptLocationForWindow(userData.window2);
               } else if (currentHour >= 18 && currentHour <= 21) {
```
Comment (eschlange) [inline app/scripts/controllers/main.js]: Just checking, but is 18 the correct hour? Is there something that should happen between  17 and 18? Same goes for the previous gap between 12 and 13.

File: app/scripts/controllers/slide.js (hunk 1 above)
Comment (ericcf) [inline app/scripts/controllers/slide.js]: Nice catch.

File: app/scripts/controllers/slide.js [hunk 3]
```diff
@@ -108,7 +112,7 @@
           intervention_step: 1,
           response: response,
           date_created: new Date(),
-          contact_type: contact_type,
+          contact_type: isPrompt ? 'Prompt' : 'On-Demand',
```
Comment (michaelwehrley) [inline app/scripts/controllers/slide.js]: Nice improvement.

File: app/scripts/controllers/main.js (hunk 2 above)
Comment (ericcf) [inline app/scripts/controllers/main.js]: Yeah, this means that 1:00pm-5:59pm will be a block and 6:00pm-9:59pm will be a block.

File: test/unit/controllers/MainControllerSpec.js [hunk 4]
```diff
@@ -33,19 +42,31 @@ describe('MainCtrl', function() {
...
+          mockLocalStorage.userData = undefined;
+          sinon.spy(mockLocation, 'path');
+          injectController($controller);
+        });
+
+        expect(mockLocation.path.calledOnce).to.be.true;
+      });
+    });
+
+    describe('when userData has content', function() {
```
Comment (michaelwehrley) [inline test/unit/controllers/MainControllerSpec.js]: Did you mean to leave this blank?

**Comment by michaelwehrley** [comment]
:+1:

File: test/unit/controllers/MainControllerSpec.js (hunk 4 above)
Comment (ericcf) [inline test/unit/controllers/MainControllerSpec.js]: Nope... :facepunch:

**Comment by eschlange** [comment]
Great job on increasing code cohesion! @michaelwehrley , are you OK with me merging this?

**Comment by michaelwehrley** [comment]
@eschlange I am, but maybe @ericcf wants to delete that `describe` block.

**Comment by ericcf** [comment]
Ok I added a spec and deleted the unused service. Merging.

**Pull request merged**